core/src/main/java/com/nerds/gamejam/gameplay/planet/PlanetFactory.java

```diff
@@ -80,7 +81,7 @@ public void createPlanet(World world, int orbitalRadius, int solarCenterX, int s
                 .add(new FontComponent(planetName, x - 10, y - 10))
                 .add(new ClickableComponent((worldX, worldY, screenX, screenY, button) -> {
                     String desc = String.format(GameJam.gameStrings.get("planetDescription"), GameJam.gameStrings.get(finalBaseMaterial.name().toLowerCase()), GameJam.gameStrings.get(finalLandmass.name().toLowerCase()), GameJam.gameStrings.get(finalSecondaryMaterial.name().toLowerCase()));
-                    worldEntity.edit().add(new SelectedPlanet(planetName, x, y, radius, desc));
+                    worldEntity.edit().add(new SelectedPlanet(planetName, positionComponent.x, positionComponent.y, radius, desc));
```

Comment:
Tbh, SelectedPlanet probably shouldn't take an X and Y if it can just use the planet's X and Y. Since we're adding it to the same component we could update the ECS system so it looks for entities with a SelectedPlanet and Position?